app/services/check_sys_admin_authorisation.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class CheckSysAdminAuthorisation
+  include ServicePattern
+
+  def initialize(session:)
+    @session = session
+  end
+
+  def call
+    signin_user =
+      if otp_feature_enabled?
+        OtpSignInUser.load_from_session(@session)
+      else
+        DfESignInUser.load_from_session(@session)
+      end
```

Comment:
might `DfESignInUser.load_from_session(@session) || OtpSignInUser.load_from_session(@session)` do the trick here?

Reply:
ah sorry, you've already had this thought!